### PR TITLE
fix parentheses around sequence expression as body of arrow chain

### LINKED ISSUE #10955: Using compound expression with comma operator as return value of curried function
**Prettier 2.3.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzArlMTAXkwAoBDAJwHMBGAGkyuoCYBKYgPjI6O4poMmNdgB0oIeiAgAHGAEt0yUFUoQA7gAUqCNMhDkANuvIBPPVIBGlcmADWcGAGUZt+VGrIYlXHClwAW0s4ABMQ0IAZcg9ccmo4ADEISgDyGAUPfXJcGAhJEAALGADDAHUC+Xg0VzA4J11K+QA3StN9MDQLEHc0OEoYTRtqVORsI16pACs0AA8AIRt7RydyALgI9zhR8b8QaZmnd2pDOABFXAh4bcMJkFdKXsp9S3Jgw3yZSncYUvkQmAKyAAHAAGKSfCC9Uo2GT6T5wR5NLZSACOF3gg1kehQ5DQAFooHBQqF8pQ4Gj5GTBnERkgxjddr0AvIvD5GUcTudLls6TspDBXr9-oCkCx+TZ5IYjgBhCABWkgBEAVnyuF6ABVXtj6bcmr4AJJQcKwJxgL5yACCRqcMFMJ2uvQAvo6gA)

```sh
--parser babel
```

**Input:**

```js
const func = (arg1, arg2) => () => (arg1, arg2)

```

**Output:**

```js
const func = (arg1, arg2) => () => arg1, arg2;

```

**Expected behavior:**
Using compound expression with comma operator is not usual, but is valid syntax (and can be really useful).
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator

It should be handled correctly by prettier in all cases, including this currying case.

Given input should result with no changes at all.
The current output is not valid JS syntax as the `arg2` variable is not declared, and this gives an error when executed.

### LINKED ISSUE #11578: Removing parentheses around comma separated values breaks the code
**Prettier 2.4.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzKmC8mAFAJQEB8xAHmfpUQAwA0mNAOlCEyBAA4wBLdMlABDAE7iIAdwAKEhGmQhRAG2miAnku4AjcaLABrODADKogLZwAMgKhxk2NWjh6Dx02d6H7Ac2QYcQBXNxA4S104ABNomJtRKD9g0T84ADEIcUtRGEEk5VFgmAguEAALGEtVAHVygXg0HzA4M0UGgQA3Bs1lMDQdEHtXcRhZAz8cpxcwgCs0KjN-VTgARWCIeGnVV24fcRHlXVEo1TLecXsYGoFomHLkAA5mEAuIVxqDXmULuBHOxzcACOG3g4z4ShQojQAFoHDEYmVxHAQQJkeNUlMkM4dmFXJYBIEQnjlmtQY5sTNuDATjc7g8kAAmakGASqfwAYQglix4TQAFYysFXAAVE6QnG7ECdUIASSgcVgZjAl34AEEFWYYJoVttXABffVAA)

```sh
--parser babel
```

**Input:**

```jsx
const fn = () => (x) => (0, x)

```

**Output:**

```jsx
const fn = () => (x) => 0, x;

```

**Expected behavior:**

Parentheses stays as they are.

### PR DESCRIPTION
**Pull request by bakkot**
## Description

Fixes #10955.

#9992 introduced a special case for printing the bodies of chained arrows, which has very different logic from printing a regular arrow. It neglected to copy over the logic to handle this case.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by bakkot** [comment]
Sidebar: is there a reason #9992 is using `line` instead of `softline`? It means that even when the chain fits on one line, as in

```js
const f = () => ({ k }) => k;
```
it will get split across multiple lines, as in
```js
const f =
  () =>
  ({ k }) =>
    k;
```

That seems obviously wrong to me; I'd be happy to submit a PR fixing it.

**Comment by fisker** [comment]
@bakkot Have you seen #11578? I think we already have code for this https://github.com/prettier/prettier/blob/20df210ddae98bf8a7631227d1fc0c8224f27f97/src/language-js/print/function.js#L343, but we returned too early .

**Comment by bakkot** [comment]
@fisker [I wrote that comment](https://github.com/prettier/prettier/pull/2388). And yes, the problem is that #9992 introduced a `return` before the logic which handles sequence expressions, as I said in the description; the fix is to have the new code handle sequence expressions as well.

**Comment by fisker** [comment]
Sorry, I didn't see the description.

I don't like coping the existing code, but seems hard to reuse?

**Review by fisker — APPROVED** [review]
LGTM, thank you!

File: changelog_unreleased/javascript/11593.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#### Title (#11593 by @bakkot)
```
Comment (sosukesuzuki) [inline changelog_unreleased/javascript/11593.md:1]:
```suggestion
#### Fix parentheses around sequence expression as body of arrow chain (#11593 by @bakkot)
```

**Review by sosukesuzuki — APPROVED** [review]
Thank you! But please update changelog.

**Pull request merged**